src/Storages/System/StorageSystemParts.cpp

```diff
@@ -235,11 +235,8 @@ void StorageSystemParts::processNextStorage(
         if (columns_mask[src_index++])
         {
             /// The full path changes at clean up thread, so do not read it if parts can be deleted, avoid the race.
-            if (part->isStoredOnDisk()
-                && part_state != State::Deleting && part_state != State::DeleteOnDestroy && part_state != State::Temporary)
-            {
+            if (part->isStoredOnDisk() && !part->isRemoveOfPartStarted())
```

Comment:
as I understand the race is still possible, but it is quite unlikely that it will happen?